packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -415,17 +450,23 @@ abstract contract CrossDomainMessenger is
      * @return Amount of gas required to guarantee message receipt.
      */
     function baseGas(bytes calldata _message, uint32 _minGasLimit) public pure returns (uint64) {
```

Comment:
IMO, we should revert in `sendMessage` if sending the result of `baseGas` is unachievable based on the constraints of the L1 block gas limit.

Reply:
Discussion in slack, but placing here for posterity: We should also consider **not** restricting this and place this logic in the sdk so future teams can adopt these contracts for various system constraints without modification and needing to engage an audit. If the change happens in the sdk, it's much more lightweight to change in the future.